org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/DOMNode.java

```diff
@@ -378,6 +378,26 @@ public void setAttributeNode(DOMAttr attr) {
 		return attributeNodes;
 	}
 
+	/**
+	 * Returns a list of children, each having an attribute called name, with a value
+	 * of value
+	 * @param name name of attribute
+	 * @param value value of attribute
+	 * @return  list of children, each having a specified attribute name and value
+	 */
+	public List<DOMNode> getChildrenWithAttributeValue(String name, String value) {
+		List<DOMNode> result = new ArrayList<>();
+		for (DOMNode child: getChildren()) {
+			if (child.hasAttribute(name)) {
+				String attrValue = child.getAttribute(name);
+				if (attrValue.equals(value)) {

```

Comment:
It should be better to do `Objects.equals(attrValue, value)` to prevent from NPE